src/components/Clients.vue

```diff
@@ -0,0 +1,108 @@
+<template>
+    <div class="container-fluid" style="width: 80%" @before-open="beforeOpen" >
+        <client-modal></client-modal>
+        <b-col>
+            <div class="row filter top-buffer">
+                <input class="search form-control form-control-lg col-9" type="text" placeholder="Chercher par client" >
+            </div>
+            <div class="row card top-buffer">
+                <div class="col-12">
+                    <table class="table table-borderless">
+                        <thead>
+                        <tr>
+                            <th scope="col">nom_client</th>
+                            <th scope="col">prenom_client</th>
+                            <th scope="col">telephone_client</th>
+                            <th scope="col">email_client</th>
+                            <th scope="col">country_client</th>
+                            <th scope="col">Actions</th>
+                        </tr>
+                        </thead>
+                        <tbody>
+                        <tr v-for="(client,index) in clients" :key="index">
+                            <th scope="row">{{client.nom_client}}</th>
+                            <th scope="row">{{client.prenom_client}}</th>
+                            <td>{{client.telephone_client}}</td>
+                            <td>{{client.email_client}}</td>
+                            <td>{{client.country_client}}</td>
+                            <td></td>
+                           <td>
+                                <button type="button" class="btn btn-primary"  @click="$modal.show('client-modal') ">R</button>
+                                <button type="button" class="btn btn-warning"  @click="$modal.show('client-modal') ">U</button>
+                                <button type="button" class="btn btn-danger"  @click="DeleteUser(client.id_client, index)">D</button>
+                            </td>
+                        </tr>
+                       </tbody>
+                    </table>
+                </div>
+            </div>
+        </b-col>
+    </div>
+</template>
+
+<script>
+    import ClientModal from "./ClientModal";
+
+    export default {
+        name: "Clients",
+        components: {
+            ClientModal
+        },
+        data(){
+            return{
+                clients:{},
+            }
+        },
+        created :function() {
+            this.beforeOpen();
+        },
+        methods:{
+            beforeOpen(){
+                console.log('before Open called');
+                this.reset();
+                Promise.all([
```

Comment:
We don't need promise, we do promise to execute something after fetch in then scope